In [141]:
%load_ext autoreload
%autoreload 2

from src.base_classes.omic_data_loader import OmicDataLoader
from src.data_managers.concat import CatOmicDataManager

The autoreload extension is already loaded. To reload it, use:
  %reload_ext autoreload


In [142]:
mrna_loader = OmicDataLoader(
    data_dir="mds_data/splits_74/mrna",
)
mirna_loader = OmicDataLoader(
    data_dir="mds_data/splits_74/mirna_genes",
)
circrna_loader = OmicDataLoader(
    data_dir="mds_data/splits_74/circrna",
)
# pirna_loader = OmicDataLoader(
#     data_dir="mds_data/splits_74/pirna",
# )
te_loader = OmicDataLoader(
    data_dir="mds_data/splits_74/te_counts",
)

In [180]:
omic_data_loaders = {
    "mrna": mrna_loader,
    "mirna": mirna_loader,
    "circrna": circrna_loader,
    # "pirna": pirna_loader,
    "te": te_loader,
}
odm = CatOmicDataManager(omic_data_loaders, n_splits=5)

save_folder = f"logs/mds_disease/{'_'.join(omic_data_loaders.keys())}.csv"
save_folder

'logs/mds_disease/mrna_mirna_circrna_te.csv'

In [144]:
from src.evals.knn import KNNEvaluator

knn_eval = KNNEvaluator(
    data_manager=odm,
    n_trials=20,
    verbose=True,
    params={"k_lb": 1, "k_ub": 20},
)
_ = knn_eval.evaluate()
knn_eval.save_results(results_file=save_folder, row_name="KNN")

[I 2024-11-17 21:05:50,513] A new study created in memory with name: no-name-874da8ca-1795-4656-a64d-fc031d195920
[I 2024-11-17 21:05:50,599] Trial 0 finished with value: 0.27794757275052406 and parameters: {'n_neighbors': 20}. Best is trial 0 with value: 0.27794757275052406.


New best score: 0.278
Best model performance:
Accuracy: 0.825 ± 0.030
F1 Macro: 0.452 ± 0.009
F1 Weighted: 0.746 ± 0.043


[I 2024-11-17 21:05:50,663] Trial 1 finished with value: 0.6528081564994511 and parameters: {'n_neighbors': 6}. Best is trial 1 with value: 0.6528081564994511.
[I 2024-11-17 21:05:50,727] Trial 2 finished with value: 0.32661842676062103 and parameters: {'n_neighbors': 13}. Best is trial 1 with value: 0.6528081564994511.
[I 2024-11-17 21:05:50,788] Trial 3 finished with value: 0.32661842676062103 and parameters: {'n_neighbors': 13}. Best is trial 1 with value: 0.6528081564994511.


New best score: 0.653
Best model performance:
Accuracy: 0.907 ± 0.053
F1 Macro: 0.806 ± 0.106
F1 Weighted: 0.893 ± 0.062


[I 2024-11-17 21:05:50,852] Trial 4 finished with value: 0.6528081564994511 and parameters: {'n_neighbors': 5}. Best is trial 1 with value: 0.6528081564994511.
[I 2024-11-17 21:05:50,913] Trial 5 finished with value: 0.46772551930590295 and parameters: {'n_neighbors': 7}. Best is trial 1 with value: 0.6528081564994511.
[I 2024-11-17 21:05:50,976] Trial 6 finished with value: 0.46772551930590295 and parameters: {'n_neighbors': 7}. Best is trial 1 with value: 0.6528081564994511.
[I 2024-11-17 21:05:51,045] Trial 7 finished with value: 0.32661842676062103 and parameters: {'n_neighbors': 13}. Best is trial 1 with value: 0.6528081564994511.
[I 2024-11-17 21:05:51,112] Trial 8 finished with value: 0.32661842676062103 and parameters: {'n_neighbors': 14}. Best is trial 1 with value: 0.6528081564994511.
[I 2024-11-17 21:05:51,177] Trial 9 finished with value: 0.4419268561846254 and parameters: {'n_neighbors': 1}. Best is trial 1 with value: 0.6528081564994511.
[I 2024-11-17 21:05:51,241] Trial 

Results saved to logs/mds_disease/te.csv


In [145]:
from src.evals.svm import SVMEvaluator

svm_eval = SVMEvaluator(
    data_manager=odm,
    n_trials=50,
    verbose=True,
    params={
        "C_lb": 0.01,
        "C_ub": 10,
        "no_rfe": True,
        # "rfe_step_range": (0.05, 0.2),
        # "rfe_n_features_range": (100, 200),
    },
    mode="linear",
)
_ = svm_eval.evaluate()
svm_eval.save_results(results_file=save_folder, row_name="Linear SVM")

[I 2024-11-17 21:05:51,874] A new study created in memory with name: no-name-3649c405-4775-425f-921e-c96a64d938fe
/home/lubojjan/DiplomaGeneral/.venv/lib/python3.12/site-packages/sklearn/svm/_base.py:1235: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  warnings.warn(
/home/lubojjan/DiplomaGeneral/.venv/lib/python3.12/site-packages/sklearn/svm/_base.py:1235: ConvergenceWarning: Liblinear failed to converge, increase the number of iterations.
  warnings.warn(
[I 2024-11-17 21:05:51,968] Trial 0 finished with value: 0.7389770960193865 and parameters: {'C': 2.175199734600259, 'class_weight': 'balanced'}. Best is trial 0 with value: 0.7389770960193865.
[I 2024-11-17 21:05:52,027] Trial 1 finished with value: 0.5614209168802691 and parameters: {'C': 0.017940740966147034, 'class_weight': 'balanced'}. Best is trial 0 with value: 0.7389770960193865.


New best score: 0.739
Best model performance:
Accuracy: 0.933 ± 0.042
F1 Macro: 0.857 ± 0.094
F1 Weighted: 0.924 ± 0.051


[I 2024-11-17 21:05:52,099] Trial 2 finished with value: 0.7389770960193865 and parameters: {'C': 0.190355706863225, 'class_weight': None}. Best is trial 0 with value: 0.7389770960193865.
[I 2024-11-17 21:05:52,173] Trial 3 finished with value: 0.5329721476347162 and parameters: {'C': 0.010173158170458907, 'class_weight': 'balanced'}. Best is trial 0 with value: 0.7389770960193865.
[I 2024-11-17 21:05:52,243] Trial 4 finished with value: 0.7389770960193865 and parameters: {'C': 0.37137764156062775, 'class_weight': None}. Best is trial 0 with value: 0.7389770960193865.
[I 2024-11-17 21:05:52,331] Trial 5 finished with value: 0.7039337563024542 and parameters: {'C': 1.0018313664643688, 'class_weight': 'balanced'}. Best is trial 0 with value: 0.7389770960193865.
[I 2024-11-17 21:05:52,411] Trial 6 finished with value: 0.6354207758053912 and parameters: {'C': 0.20756595151118823, 'class_weight': 'balanced'}. Best is trial 0 with value: 0.7389770960193865.
[I 2024-11-17 21:05:52,496] Trial 

In [146]:
from src.evals.xgboost import XGBoostEvaluator

xgb_eval = XGBoostEvaluator(
    data_manager=odm,
    n_trials=60,
    verbose=True,
)
_ = xgb_eval.evaluate()
xgb_eval.print_best_results()
xgb_eval.save_results(results_file=save_folder, row_name="XGBoost")

[I 2024-11-17 21:05:56,125] A new study created in memory with name: no-name-d099f064-6173-4ef4-a5bf-4f86ad7245ed
[I 2024-11-17 21:05:56,196] Trial 0 finished with value: 0.7389770960193865 and parameters: {'lambda': 7.608655845700436e-05, 'alpha': 0.0191210546499596}. Best is trial 0 with value: 0.7389770960193865.
[I 2024-11-17 21:05:56,270] Trial 1 finished with value: 0.7389770960193865 and parameters: {'lambda': 0.01119519855589457, 'alpha': 1.9342855459957053e-06}. Best is trial 0 with value: 0.7389770960193865.
[I 2024-11-17 21:05:56,343] Trial 2 finished with value: 0.7039337563024542 and parameters: {'lambda': 0.0018460422588911568, 'alpha': 1.1667913504568424e-05}. Best is trial 0 with value: 0.7389770960193865.


New best score: 0.739
Best model performance:
Accuracy: 0.933 ± 0.042
F1 Macro: 0.857 ± 0.094
F1 Weighted: 0.924 ± 0.051


[I 2024-11-17 21:05:56,431] Trial 3 finished with value: 0.7389770960193865 and parameters: {'lambda': 0.01831112180093915, 'alpha': 3.4934732637298096e-08}. Best is trial 0 with value: 0.7389770960193865.
[I 2024-11-17 21:05:56,510] Trial 4 finished with value: 0.7581635155117249 and parameters: {'lambda': 7.155342168382797e-06, 'alpha': 0.00011441997432654826}. Best is trial 4 with value: 0.7581635155117249.
[I 2024-11-17 21:05:56,578] Trial 5 finished with value: 0.7948768203234714 and parameters: {'lambda': 1.2698271656565215e-08, 'alpha': 4.713328847699792e-05}. Best is trial 5 with value: 0.7948768203234714.
[I 2024-11-17 21:05:56,645] Trial 6 finished with value: 0.27794757275052406 and parameters: {'lambda': 2.8174953641681496e-07, 'alpha': 0.3576092545055356}. Best is trial 5 with value: 0.7948768203234714.


New best score: 0.758
Best model performance:
Accuracy: 0.933 ± 0.060
F1 Macro: 0.877 ± 0.114
F1 Weighted: 0.927 ± 0.067
New best score: 0.795
Best model performance:
Accuracy: 0.947 ± 0.050
F1 Macro: 0.894 ± 0.106
F1 Weighted: 0.939 ± 0.060


[I 2024-11-17 21:05:56,715] Trial 7 finished with value: 0.7021274450808679 and parameters: {'lambda': 0.004304511530976998, 'alpha': 0.024498026054953062}. Best is trial 5 with value: 0.7948768203234714.
[I 2024-11-17 21:05:56,784] Trial 8 finished with value: 0.3908956158579512 and parameters: {'lambda': 0.003326260439943442, 'alpha': 0.08275031292710416}. Best is trial 5 with value: 0.7948768203234714.
[I 2024-11-17 21:05:56,855] Trial 9 finished with value: 0.8011193651582444 and parameters: {'lambda': 4.018664776877043e-07, 'alpha': 2.8629988909277157e-06}. Best is trial 9 with value: 0.8011193651582444.
[I 2024-11-17 21:05:56,929] Trial 10 finished with value: 0.7948768203234714 and parameters: {'lambda': 1.5390845083711636e-06, 'alpha': 1.5047986188478118e-08}. Best is trial 9 with value: 0.8011193651582444.
[I 2024-11-17 21:05:57,008] Trial 11 finished with value: 0.7389770960193865 and parameters: {'lambda': 1.0202475155720426e-08, 'alpha': 0.0003258510791609648}. Best is tria

New best score: 0.801
Best model performance:
Accuracy: 0.947 ± 0.050
F1 Macro: 0.900 ± 0.106
F1 Weighted: 0.941 ± 0.059


[I 2024-11-17 21:05:57,081] Trial 12 finished with value: 0.7581635155117249 and parameters: {'lambda': 1.1638802462134763e-08, 'alpha': 1.0789669151809172e-06}. Best is trial 9 with value: 0.8011193651582444.
[I 2024-11-17 21:05:57,154] Trial 13 finished with value: 0.7021274450808679 and parameters: {'lambda': 1.7127788954939889e-07, 'alpha': 0.0011876355243646036}. Best is trial 9 with value: 0.8011193651582444.
[I 2024-11-17 21:05:57,230] Trial 14 finished with value: 0.7948768203234714 and parameters: {'lambda': 1.2747083682342178e-07, 'alpha': 1.0192153993425504e-05}. Best is trial 9 with value: 0.8011193651582444.
[I 2024-11-17 21:05:57,310] Trial 15 finished with value: 0.5239116671839724 and parameters: {'lambda': 0.777806727228329, 'alpha': 2.0375054016490454e-07}. Best is trial 9 with value: 0.8011193651582444.
[I 2024-11-17 21:05:57,390] Trial 16 finished with value: 0.7389770960193865 and parameters: {'lambda': 1.881372941266466e-05, 'alpha': 0.0017416820267313622}. Best i

New best score: 0.840
Best model performance:
Accuracy: 0.960 ± 0.053
F1 Macro: 0.918 ± 0.113
F1 Weighted: 0.953 ± 0.064


[I 2024-11-17 21:05:58,700] Trial 33 finished with value: 0.8011193651582444 and parameters: {'lambda': 1.0274434301778616e-05, 'alpha': 5.9039717014673634e-06}. Best is trial 30 with value: 0.8403990328047336.
[I 2024-11-17 21:05:58,805] Trial 34 finished with value: 0.7948768203234714 and parameters: {'lambda': 5.4579666059930605e-05, 'alpha': 1.1177628817652933e-06}. Best is trial 30 with value: 0.8403990328047336.
[I 2024-11-17 21:05:58,878] Trial 35 finished with value: 0.8403990328047336 and parameters: {'lambda': 6.633884268833129e-06, 'alpha': 5.963075822354323e-06}. Best is trial 30 with value: 0.8403990328047336.
[I 2024-11-17 21:05:58,952] Trial 36 finished with value: 0.7948768203234714 and parameters: {'lambda': 0.0005809140957209196, 'alpha': 6.732344835858891e-08}. Best is trial 30 with value: 0.8403990328047336.
[I 2024-11-17 21:05:59,031] Trial 37 finished with value: 0.7948768203234714 and parameters: {'lambda': 2.982735105384673e-06, 'alpha': 2.711817838521352e-06}. 

Best model performance:
Accuracy: 0.960 ± 0.053
F1 Macro: 0.918 ± 0.113
F1 Weighted: 0.953 ± 0.064


In [147]:
from src.evals.mlp import MLPEvaluator

mlp_eval = MLPEvaluator(
    data_manager=odm,
    n_trials=10,
    verbose=True,
    params={
        "lr_range": [1e-4, 1e-2],
        "l2_lambda": 5e-4,
        "dropout_range": [0.1, 0.6],
        "hidden_channels": 64,
        "proj_dim": 64,
        "batch_size": 32,
        "max_epochs": 50,
    },
)
_ = mlp_eval.evaluate()
mlp_eval.save_results(results_file=save_folder, row_name="MLP")

[I 2024-11-17 21:06:00,763] A new study created in memory with name: no-name-4058343b-a46b-4637-b1d3-c0065d3be176
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs
/home/lubojjan/DiplomaGeneral/.venv/lib/python3.12/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:424: The 'train_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=11` in the `DataLoader` to improve performance.
/home/lubojjan/DiplomaGeneral/.venv/lib/python3.12/site-packages/pytorch_lightning/loops/fit_loop.py:298: The number of training batches (2) is smaller than the logging interval Trainer(log_every_n_steps=50). Set a lower value for log_every_n_steps if you want to see logs for the training epoch.
`Trainer.fit` stopped: `max_epochs=50` reached.
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
HPU available: False, using:

New best score: 0.327
Best model performance:
Accuracy: 0.838 ± 0.031
F1 Macro: 0.505 ± 0.103
F1 Weighted: 0.771 ± 0.052


`Trainer.fit` stopped: `max_epochs=50` reached.
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs
/home/lubojjan/DiplomaGeneral/.venv/lib/python3.12/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:424: The 'train_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=11` in the `DataLoader` to improve performance.
/home/lubojjan/DiplomaGeneral/.venv/lib/python3.12/site-packages/pytorch_lightning/loops/fit_loop.py:298: The number of training batches (2) is smaller than the logging interval Trainer(log_every_n_steps=50). Set a lower value for log_every_n_steps if you want to see logs for the training epoch.
`Trainer.fit` stopped: `max_epochs=50` reached.
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs
/home/lubojjan/DiplomaGeneral/.venv/lib/python3.12/site-pa

New best score: 0.704
Best model performance:
Accuracy: 0.920 ± 0.050
F1 Macro: 0.840 ± 0.097
F1 Weighted: 0.911 ± 0.056


`Trainer.fit` stopped: `max_epochs=50` reached.
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs
/home/lubojjan/DiplomaGeneral/.venv/lib/python3.12/site-packages/pytorch_lightning/trainer/connectors/data_connector.py:424: The 'train_dataloader' does not have many workers which may be a bottleneck. Consider increasing the value of the `num_workers` argument` to `num_workers=11` in the `DataLoader` to improve performance.
/home/lubojjan/DiplomaGeneral/.venv/lib/python3.12/site-packages/pytorch_lightning/loops/fit_loop.py:298: The number of training batches (2) is smaller than the logging interval Trainer(log_every_n_steps=50). Set a lower value for log_every_n_steps if you want to see logs for the training epoch.
`Trainer.fit` stopped: `max_epochs=50` reached.
GPU available: False, used: False
TPU available: False, using: 0 TPU cores
HPU available: False, using: 0 HPUs
/home/lubojjan/DiplomaGeneral/.venv/lib/python3.12/site-pa

In [148]:
mlp_eval.print_best_results()
mlp_eval.print_best_parameters()

Best model performance:
Accuracy: 0.920 ± 0.050
F1 Macro: 0.840 ± 0.097
F1 Weighted: 0.911 ± 0.056
Best hyperparameters:
{'lr': 0.002579583955973857, 'dropout': 0.47167302280825596}


In [ ]:
from src.evals.mogonet import MOGONETEvaluator
from src.data_managers.sample_graph import SampleGraphDataManager

# omic_data_loaders = {
#     "mrna": mrna_loader,
#     # "mirna": mirna_loader,
#     # "circrna": circrna_loader,
#     # # "pirna": pirna_loader,
#     # "te": te_loader,
# }

mogonet_eval = MOGONETEvaluator(
    data_manager=SampleGraphDataManager(
        omic_data_loaders=omic_data_loaders,
        n_splits=5,
        params={
            "graph_style": "threshold",
            "self_connections": True,
            "avg_degree": 5,
        },
    ),
    n_trials=1,
    params={
        "encoder_hidden_channels": {
            "mrna": 64,
            "mirna": 64,
            "circrna": 64,
            "te": 64,
        },
        "encoder_type": "gat",
        "dropout": 0.2,
        "integrator_type": "vcdn", # vcdn seems to work better on mds disease
        "integration_in_dim": 16,
        "vcdn_hidden_channels": 16,
        "epochs": 250,
        "log_interval": 251,
    }
)
mogonet_eval.evaluate()
mogonet_eval.print_best_results()

save_folder = f"logs/mds_disease/{'_'.join(omic_data_loaders.keys())}.csv"
mogonet_eval.save_results(results_file=save_folder, row_name="MOGONET")

[I 2024-11-17 21:35:01,979] A new study created in memory with name: no-name-c45dede4-3987-4583-8217-5ebdaed1759e


Using: vcdn integrator
Using: vcdn integrator
Using: vcdn integrator
Using: vcdn integrator
Using: vcdn integrator


[I 2024-11-17 21:35:12,454] Trial 0 finished with value: 0.8601248426666666 and parameters: {}. Best is trial 0 with value: 0.8601248426666666.


New best score: 0.860
Best model performance:
Accuracy: 0.960 ± 0.053
F1 Macro: 0.934 ± 0.085
F1 Weighted: 0.959 ± 0.054
Best model performance:
Accuracy: 0.960 ± 0.053
F1 Macro: 0.934 ± 0.085
F1 Weighted: 0.959 ± 0.054


In [167]:
save_folder = f"logs/mds_disease/{'_'.join(omic_data_loaders.keys())}.csv"
mogonet_eval.save_results(results_file=save_folder, row_name="MOGONET")

# linear
Accuracy: 0.947 ± 0.050
F1 Macro: 0.910 ± 0.080
F1 Weighted: 0.945 ± 0.051
# vcdn
- integration dim = 2
Accuracy: 0.825 ± 0.030
F1 Macro: 0.452 ± 0.009
F1 Weighted: 0.746 ± 0.043
- integration dim = 8
Accuracy: 0.973 ± 0.033
F1 Macro: 0.952 ± 0.059
F1 Weighted: 0.971 ± 0.035
- integration dim = 12
Accuracy: 0.973 ± 0.033
F1 Macro: 0.957 ± 0.053
F1 Weighted: 0.973 ± 0.033
- integration dim = 16
Accuracy: 0.973 ± 0.053
F1 Macro: 0.958 ± 0.083
F1 Weighted: 0.973 ± 0.053
# attention - faster than vcdn
- integration dim = 2
Accuracy: 0.933 ± 0.060
F1 Macro: 0.877 ± 0.114
F1 Weighted: 0.927 ± 0.067
- integration dim = 8
Accuracy: 0.973 ± 0.033
F1 Macro: 0.952 ± 0.059
F1 Weighted: 0.971 ± 0.035
- integration dim = 12
Accuracy: 0.947 ± 0.050
F1 Macro: 0.910 ± 0.080
F1 Weighted: 0.945 ± 0.051
- integration dim = 16
Accuracy: 0.960 ± 0.053
F1 Macro: 0.934 ± 0.085
F1 Weighted: 0.959 ± 0.054

In [ ]:
from src.evals.birgat import BiRGATEvaluator
from src.data_managers.bipartite_graph import BipartiteGraphDataManager

birgat_eval = BiRGATEvaluator(
    data_manager=BipartiteGraphDataManager(
        omic_data_loaders=omic_data_loaders,
        n_splits=5,
        params={
            "diff_exp_thresholds" : {
                "mrna": 1.5,
                "mirna": 1.5,
                "circrna": 1.7,
                "te": 1.7,
            },
        },
    ),
    params={
        "epochs": 350,
        "log_interval": 50,
        "hidden_channels": [200, 64, 64, 16, 16],
        "heads": 4,
        "dropout": 0.2,
        "attention_dropout": 0.0,
        "integrator_type": "vcdn",
        "three_layers": True,
    },
    n_trials=1,
)
birgat_eval.evaluate()
birgat_eval.print_best_results()
# birgat_eval.save_results(results_file=save_folder, row_name="BiRGAT")

isolated sample nodes, isolated gene nodes, mean degree: 
tensor(1) tensor(0) tensor(25.6892)
isolated sample nodes, isolated gene nodes, mean degree: 
tensor(0) tensor(0) tensor(21.1486)
isolated sample nodes, isolated gene nodes, mean degree: 
tensor(0) tensor(9) tensor(16.1351)
isolated sample nodes, isolated gene nodes, mean degree: 
tensor(4) tensor(0) tensor(15.5676)
isolated sample nodes, isolated gene nodes, mean degree: 
tensor(1) tensor(0) tensor(25.6892)
isolated sample nodes, isolated gene nodes, mean degree: 
tensor(0) tensor(0) tensor(21.1486)


[I 2024-11-17 21:42:15,921] A new study created in memory with name: no-name-c968c791-c334-4a18-8f15-8e7f34cbb609


isolated sample nodes, isolated gene nodes, mean degree: 
tensor(0) tensor(9) tensor(16.1351)
isolated sample nodes, isolated gene nodes, mean degree: 
tensor(4) tensor(0) tensor(15.5676)
isolated sample nodes, isolated gene nodes, mean degree: 
tensor(1) tensor(0) tensor(25.6892)
isolated sample nodes, isolated gene nodes, mean degree: 
tensor(0) tensor(0) tensor(21.1486)
isolated sample nodes, isolated gene nodes, mean degree: 
tensor(0) tensor(9) tensor(16.1351)
isolated sample nodes, isolated gene nodes, mean degree: 
tensor(4) tensor(0) tensor(15.5676)

Epoch: 050:
Train Loss: 0.4453, Train Acc: 0.7966, Train F1 Macro: 0.4434, Train F1 Weighted: 0.7365
Val Acc: 0.8000, Val F1 Macro: 0.4444, Val F1 Weighted: 0.7111, Val Geometric Mean: 0.6323
Test Acc: 0.8000, Test F1 Macro: 0.4444, Test F1 Weighted: 0.7111
##################################################

Epoch: 100:
Train Loss: 0.1416, Train Acc: 0.9322, Train F1 Macro: 0.8554, Train F1 Weighted: 0.9251
Val Acc: 0.8667, Val F1 

[I 2024-11-17 21:45:09,700] Trial 0 finished with value: 0.8403990328047336 and parameters: {}. Best is trial 0 with value: 0.8403990328047336.



Epoch: 350:
Train Loss: 0.1649, Train Acc: 0.9333, Train F1 Macro: 0.8958, Train F1 Weighted: 0.9354
Val Acc: 0.7143, Val F1 Macro: 0.5758, Val F1 Weighted: 0.7489, Val Geometric Mean: 0.6753
Test Acc: 0.7143, Test F1 Macro: 0.5758, Test F1 Weighted: 0.7489
##################################################
New best score: 0.840
Best model performance:
Accuracy: 0.960 ± 0.053
F1 Macro: 0.918 ± 0.113
F1 Weighted: 0.953 ± 0.064
Best model performance:
Accuracy: 0.960 ± 0.053
F1 Macro: 0.918 ± 0.113
F1 Weighted: 0.953 ± 0.064


In [160]:
birgat_eval.save_results(results_file=save_folder, row_name="BiRGAT")

```python
omic_data_loaders = {
    "mrna": mrna_loader,
    "mirna": mirna_loader,
    "circrna": circrna_loader,
    # # "pirna": pirna_loader,
    # "te": te_loader,
}

birgat_eval = BiRGATEvaluator(
    data_manager=BipartiteGraphDataManager(
        omic_data_loaders=omic_data_loaders,
        n_splits=5,
        params={
            "diff_exp_thresholds" : {
                "mrna": 1.5,
                "mirna": 1.5,
                "circrna": 1.7,
                # "te": 1.8,
            },
        },
    ),
    params={
        "epochs": 250,
        "log_interval": 50,
        "hidden_channels": [200, 64, 64, 16, 16],
        "heads": 4,
        "dropout": 0.2,
        "attention_dropout": 0.0,
        "integrator_type": "vcdn",
        "three_layers": False,
    },
    n_trials=1,
)
```

Accuracy: 0.960 ± 0.053
F1 Macro: 0.940 ± 0.082
F1 Weighted: 0.961 ± 0.053

# mrna, mirna, circrna, 2L
Accuracy: 0.960 ± 0.033
F1 Macro: 0.915 ± 0.073
F1 Weighted: 0.956 ± 0.036
# mrna, mirna, circrna, 2L no interactions
Accuracy: 0.960 ± 0.053
F1 Macro: 0.918 ± 0.113
F1 Weighted: 0.953 ± 0.064
---
Accuracy: 0.946 ± 0.027
F1 Macro: 0.904 ± 0.048
F1 Weighted: 0.944 ± 0.028
# mrna, mirna, circrna 3L, interactions, degree ~20 in diff exp graphs, larger degree shows degraded performance
# making the avg degree to high shows large jumps on the validation set during training
Accuracy: 0.945 ± 0.053
F1 Macro: 0.910 ± 0.081
F1 Weighted: 0.946 ± 0.048
# mrna, mirna, circrna 2L, interactions, 64 cap
Accuracy: 0.960 ± 0.053
F1 Macro: 0.940 ± 0.082
F1 Weighted: 0.961 ± 0.053
# mrna, mirna, circrna 3L, interactions, 64 cap
Accuracy: 0.891 ± 0.054
F1 Macro: 0.801 ± 0.088
F1 Weighted: 0.888 ± 0.056
# mrna, mirna, circrna 3L
Accuracy: 0.920 ± 0.050
F1 Macro: 0.829 ± 0.112
F1 Weighted: 0.907 ± 0.062
# mrna, mirna, 2L
Accuracy: 0.960 ± 0.033
F1 Macro: 0.915 ± 0.073
F1 Weighted: 0.956 ± 0.036
# mrna, mirna, 3L
Accuracy: 0.947 ± 0.050
F1 Macro: 0.897 ± 0.089
F1 Weighted: 0.944 ± 0.051